src/PolyType.SourceGenerator/SourceFormatter/SourceFormatter.Object.cs

```diff
@@ -39,6 +41,53 @@ private void FormatObjectTypeShapeFactory(SourceWriter writer, string methodName
         FormatMemberAccessors(writer, objectShapeModel);
     }
 
+    private static string FormatAssociatedTypesCloser(ObjectShapeModel objectShapeModel)
+    {
+        if (objectShapeModel.AssociatedTypes.Length == 0)
+        {
+            return "null";
+        }
+
+        SourceWriter writer = new() { Indentation = 0 };
+        writer.WriteLine(/* lang=c#-test */ """
```

Comment:
What incantation is this?

Reply:
Oh awesome! Should we just put the attribute on the method itself? Or does it not work well when we're emitting partial code fragments?

Reply:
It works quite well for fragments too in my experience. I'll be happy to apply that change.

Reply:
FYI it has no effect on `$"` interpolated strings, which it appears _most_ of your callers are.

Reply:
`SourceWriter` is awesome. I've already copied it into two other projects I maintain.